### PR TITLE
Update motoko examples to use jest

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/motoko_examples/phone-book/src/declarations/phone_book/index.d.ts
Comment (bdemann) [inline examples/motoko_examples/phone-book/src/declarations/phone_book/index.d.ts:1]: Why do we have the declaration files for superheroes and phone_book in the repo? Is that intentional?

File: examples/motoko_examples/counter/test/tests.ts [hunk 1]
```diff
@@ -1,60 +1,35 @@
...
-                };
-            }
-        },
-        {
-            name: 'inc',
-            test: async () => {
-                const result = await counterCanister.inc();
+            expect(result).toBe(undefined);
+        });
+        it('increments the counter', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/counter/test/tests.ts:24]: "increments the counter a second time"

File: examples/motoko_examples/http_counter/test/tests.ts [hunk 2]
```diff
@@ -1,78 +1,45 @@
...
+            expect(await getCountGzip()).toBe('query');
+        });
+
+        it('streams the value of the counter over http', async () => {
+            expect(await getCountStream()).toBe(
+                getExpectedGetCountStreamResult(3)
+            );
+        });
+
+        it('gets the final value of the counter', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/http_counter/test/tests.ts:39]: add "via http"?

File: examples/motoko_examples/quicksort/test/tests.ts [hunk 3]
```diff
@@ -1,78 +1,42 @@
...
 import { _SERVICE } from './dfx_generated/quicksort/quicksort.did';
 
-export function getTests(quicksortCanister: ActorSubclass<_SERVICE>): Test[] {
-    return [
-        {
-            name: 'sort - with values from the motoko repo',
-            test: async () => {
-                const input = [5n, 3n, 0n, 9n, 8n, 2n, 1n, 4n, 7n, 6n];
-                const expectedValues = [0n, 1n, 2n, 3n, 4n, 5n, 6n, 7n, 8n, 9n];
+export function getTests(quicksortCanister: ActorSubclass<_SERVICE>): Test {
```
Comment (bdemann) [inline examples/motoko_examples/quicksort/test/tests.ts:8]: update this to use describe.each

File: examples/motoko_examples/superheroes/test/tests.ts [hunk 4]
```diff
@@ -1,123 +1,86 @@
...
-                const superman: Superhero = {
-                    name: 'Superman',
-                    superpowers: [
-                        [
-                            'superhuman strength',
-                            [['flight', [['x-ray vision', []]]]]
-                        ]
+export function getTests(superheroesCanister: ActorSubclass<_SERVICE>): Test {
+    return () => {
+        it('creates hero without powers', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/superheroes/test/tests.ts:14]: "creates a hero without powers"

File: examples/motoko_examples/superheroes/test/tests.ts [hunk 5]
```diff
@@ -1,123 +1,86 @@
...
+            const result = await superheroesCanister.create(SPIDER_MAN);
+
+            expect(result).toBe(0);
+        });
+
+        it('creates hero with powers', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/superheroes/test/tests.ts:20]: "creates a hero with powers"

File: examples/motoko_examples/whoami/test/tests.ts [hunk 6]
```diff
@@ -13,95 +13,69 @@ function createIdentity(seed: number): SignIdentity {
...
+            expect(result).toStrictEqual(installationPrincipal);
+        });
+
+        it('gets the principal that was provided as an init argument', async () => {
+            const result = await whoamiCanister.argument();
+
+            expect(result).toStrictEqual(someonePrincipal);
+        });
+
+        it('gets the principal of caller of this method', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/whoami/test/tests.ts:45]: "gets the principal of the caller of this method"

File: examples/motoko_examples/whoami/test/tests.ts [hunk 7]
```diff
@@ -13,95 +13,69 @@ function createIdentity(seed: number): SignIdentity {
...
+            expect(result).toStrictEqual(callingPrincipal);
+        });
+
+        it('gets the principal of this canister via cross canister call', async () => {
+            const result = await whoamiCanister.id();
+
+            expect(result).toStrictEqual(canisterPrincipal);
+        });
+
+        it('gets teh principal of this canister via ic object', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/whoami/test/tests.ts:57]: "gets the principal of this cansiter via ic object"

File: examples/motoko_examples/whoami/test/tests.ts [hunk 8]
```diff
@@ -13,95 +13,69 @@ function createIdentity(seed: number): SignIdentity {
...
+        please('redeploy the canister', async () => {
+            execSync(
+                `dfx deploy --upgrade-unchanged ${canisterName} --argument '(principal "${callingPrincipal}")'`,
+                {
+                    stdio: 'inherit'
+                }
+            );
+        });
+
+        it('gets the principal that was provided as an postUpgrade argument', async () => {
```
Comment (bdemann) [inline examples/motoko_examples/whoami/test/tests.ts:72]: "gets the principal that was provided as a postUpgrade argument"

**Review by lastmjs — APPROVED** [review]

**Pull request merged**